### PR TITLE
Add CMSG_* "macro" support to Fuchsia

### PR DESCRIPTION
**Pull request by amanda-tait**
This change defines and implements functions for the Fuchsia platform
corresponding to the C library CMSG_* macros, used for processing socket
control messages sent or received using the recv_msg(2)/send_msg(2)
syscalls.

### CODE REVIEW

File: src/fuchsia/mod.rs [hunk 1]
```diff
@@ -3250,6 +3250,60 @@ f! {
         dev |= (minor & 0xffffff00) << 12;
         dev
     }
+
+    pub fn CMSG_DATA(cmsg: *const cmsghdr) -> *mut c_uchar {
+        cmsg.offset(1) as *mut c_uchar
+    }
+
+    pub fn CMSG_NXTHDR(mhdr: *const msghdr, cmsg: *const cmsghdr) ->
+        *mut cmsghdr {
```
Comment (JohnTitor) [inline src/fuchsia/mod.rs:3259]: Hm this indent is unusual, should be:
```suggestion
    pub fn CMSG_NXTHDR(mhdr: *const msghdr, cmsg: *const cmsghdr)
        -> *mut cmsghdr
    {
```

File: src/fuchsia/mod.rs [hunk 2]
```diff
@@ -3250,6 +3250,60 @@ f! {
...
+        if ((*cmsg).cmsg_len as ::size_t) < ::mem::size_of::<cmsghdr>() {
+            core::ptr::null_mut()
```
Comment (JohnTitor) [inline src/fuchsia/mod.rs:3261]: I'd prefer:

```suggestion
            0 as *mut cmsghdr
```

File: src/fuchsia/mod.rs [hunk 3]
```diff
@@ -3250,6 +3250,60 @@ f! {
...
+        } else if __CMSG_NEXT(cmsg).add(::mem::size_of::<cmsghdr>())
+            >= __MHDR_END(mhdr) {
+            core::ptr::null_mut()
```
Comment (JohnTitor) [inline src/fuchsia/mod.rs:3264]:
```suggestion
            0 as *mut cmsghdr
```

File: src/fuchsia/mod.rs [hunk 4]
```diff
@@ -3250,6 +3250,60 @@ f! {
...
+        } else {
+            __CMSG_NEXT(cmsg).cast()
+        }
+    }
+
+    pub fn CMSG_FIRSTHDR(mhdr: *const msghdr) -> *mut cmsghdr {
+        if (*mhdr).msg_controllen as ::size_t >= ::mem::size_of::<cmsghdr>() {
+            (*mhdr).msg_control.cast()
+        } else {
+            core::ptr::null_mut()
```
Comment (JohnTitor) [inline src/fuchsia/mod.rs:3274]:
```suggestion
            0 as *mut cmsghdr
```

File: src/fuchsia/mod.rs (hunk 1 above)
Comment (amanda-tait) [reply src/fuchsia/mod.rs]: Done

File: src/fuchsia/mod.rs (hunk 2 above)
Comment (amanda-tait) [reply src/fuchsia/mod.rs]: Done

File: src/fuchsia/mod.rs (hunk 3 above)
Comment (amanda-tait) [reply src/fuchsia/mod.rs]: Done

File: src/fuchsia/mod.rs (hunk 4 above)
Comment (amanda-tait) [reply src/fuchsia/mod.rs]: Done

**Comment by amanda-tait** [comment]
I've addressed your comments. Thank you for the code review!

**Pull request merged**